web_src/js/features/common-issue.js

```diff
@@ -2,15 +2,25 @@ import $ from 'jquery';
 import {updateIssuesMeta} from './repo-issue.js';
 
 export function initCommonIssue() {
-  $('.issue-checkbox').on('click', () => {
-    const numChecked = $('.issue-checkbox').children('input:checked').length;
-    if (numChecked > 0) {
+  const checkboxOperate = () => {
+    if ($('.issue-checkbox input').is(':checked')) {
       $('#issue-filters').addClass('hide');
       $('#issue-actions').removeClass('hide');
+      $('#issue-actions .six').prepend($('.issue-checkbox-all'));
     } else {
       $('#issue-filters').removeClass('hide');
       $('#issue-actions').addClass('hide');
+      $('#issue-filters .six').prepend($('.issue-checkbox-all'));
     }
+  };
+
+  $('.issue-checkbox').on('click', checkboxOperate);
+
+  $('.issue-checkbox-all').on('click', (e) => {
+    const selected = $('.issue-checkbox input:checked');
+    $('.issue-checkbox input:not(:checked)').prop('checked', 1);
+    selected.prop('checked', 0);
+    checkboxOperate(e);
```

Comment:
If I understand correctly, the code does invert selection while the name is `issue-checkbox-all` (sounds like check all), it's a little misleading.

Reply:
This can be clicked on the child checkbox in the checked all checkbox can invert the selection of the child
Any suggestions on function names?

Reply:
I think it can be called `issue-checkbox-invert`.

I will add some comments here and approve & merge.

Reply:
`issue-checkbox-invert` this is function name? or `issueCheckboxInvert` ?

Reply:
After some testing, I feel that the `all/invert` behavior is still a little confusing. For example, we can get a UI like below.

I do not think these are common UIs for end users, the `all/invert` checkbox's state is confusing. Maybe the `all/invert` element should be replaced by a icon button, without state.

![image](https://user-images.githubusercontent.com/2114189/181425525-14a5f660-e4dc-4670-bf4f-bd79055cfa32.png)

![image](https://user-images.githubusercontent.com/2114189/181425688-7cdb9946-67dc-49d0-81c4-26cb86838b99.png)


![image](https://user-images.githubusercontent.com/2114189/181425727-66476b1b-e11f-4c52-a619-68b8d12f62f7.png)

![image](https://user-images.githubusercontent.com/2114189/181425779-2f0015f2-0c31-4e9a-be4b-d7705f0f7a98.png)

Reply:
To make UI consistent, I think there could be 2 choices:

* Make the `checkbox-all` really do check all/check none
* Use a `checkbox-invert` button (without state) to toggle/invert selection

Personally I think choice 1 (check all/check none) is clear and still easy to use. How do you think?

Reply:
You mean don't do invert selection, just do select all and deselect all?

Reply:
yes

Reply:
Is it like this? https://github.com/go-gitea/gitea/pull/20177/commits/01f3f0f14d5a45fee9f1452e9469a955b1efecf2

Reply:
Thank you ~~ I also do some improvments. Now the checkbox states are synchronized and the select-all checkbox has an indeterminate state.

![image](https://user-images.githubusercontent.com/2114189/181451296-dd4a72a6-1e88-42aa-9934-5d859429ca66.png)